# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# Filename: package/.../misdn/mISDN-split-install-init.patch
# Copyright (C) 2006 The OpenSDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

mISDN-split-install-init.patch

This patch splits the installation of the init script
from the install target to it's own.

2006-08-31 Christian Wiese <cw@ixplanet.de>

--- ./Makefile.orig	2006-08-31 15:11:04.000000000 +0200
+++ ./Makefile	2006-08-31 15:14:23.000000000 +0200
@@ -53,12 +53,6 @@
 	cd $(LINUX) ; make INSTALL_MOD_PATH=$(INSTALL_PREFIX) SUBDIRS=$(MISDN_SRC) modules_install 
 	mkdir -p $(INSTALL_PREFIX)/usr/include/linux/
 	cp $(MISDNDIR)/include/linux/*.h $(INSTALL_PREFIX)/usr/include/linux/
-	mkdir -p $(INSTALL_PREFIX)/usr/sbin/
-	install -m755 misdn-init $(INSTALL_PREFIX)/usr/sbin/
-	if [ -d $(INSTALL_PREFIX)/etc/init.d ]; then \
-		if [ -e $(INSTALL_PREFIX)/etc/init.d/misdn-init ]; then rm -rf $(INSTALL_PREFIX)/etc/init.d/misdn-init; fi; \
-		ln -s $(INSTALL_PREFIX)/usr/sbin/misdn-init $(INSTALL_PREFIX)/etc/init.d/misdn-init; \
-	fi
 	mkdir -p $(INSTALL_PREFIX)/etc/modprobe.d
 	cp mISDN.modprobe.d $(INSTALL_PREFIX)/etc/modprobe.d/mISDN
 	mkdir -p $(INSTALL_PREFIX)/etc/modules.d
@@ -67,6 +61,14 @@
 	$(UPDATE_MODULES)
 	$(MODULES_UPDATE)
 
+install-init:
+	mkdir -p $(INSTALL_PREFIX)/usr/sbin/
+	install -m755 misdn-init $(INSTALL_PREFIX)/usr/sbin/
+	if [ -d $(INSTALL_PREFIX)/etc/init.d ]; then \
+		if [ -e $(INSTALL_PREFIX)/etc/init.d/misdn-init ]; then rm -rf $(INSTALL_PREFIX)/etc/init.d/misdn-init; fi; \
+		ln -s $(INSTALL_PREFIX)/usr/sbin/misdn-init $(INSTALL_PREFIX)/etc/init.d/misdn-init; \
+	fi
+
 test_old_misdn:
 	@if echo -ne "#include <linux/mISDNif.h>" | gcc -C -E - 2>/dev/null 1>/dev/null  ; then \
 		if ! echo -ne "#include <linux/mISDNif.h>\n#if MISDN_MAJOR_VERSION < 4\n#error old mISDNif.h\n#endif\n" | gcc -C -E - 2>/dev/null 1>/dev/null ; then \
